### PR TITLE
Product tax

### PR DESCRIPTION
**Pull request by vivekshekar**
@abhikchakraborty 

Product tax value enhancement.

### CODE REVIEW

File: controllers/products.js [hunk 1]
```diff
@@ -117,32 +118,92 @@ module.exports = function(app, config) {
...
+								if (k === 'tax_values' || k === 'shipping_handling_tax_values') {
+									v = common.parseTaxData(v);
+									productData[k] = v;
+								}
+								// prefixing the SO number with the SO prefix from setting
+								if (productPrefix && productPrefix.settingData && k === 'product_number') {
+									v= productPrefix.settingData+''+v;
+									productData[k] = v;
+								}
+							});
```
Comment (sQcrm) [inline controllers/products.js:162]: @vivekshekar 
- Product has only `tax_value` and there is no other tax associated.
- So check only for `tax_value`
- Product does not have customized product number so we do not need that formatting part
- May be rename the function as `setProductTaxValues`

File: controllers/products.js [hunk 2]
```diff
@@ -117,32 +118,92 @@ module.exports = function(app, config) {
...
+							return modifiedCallBack();
+						},
+						
+						// set the line items which is one-to-many relation with invoice
+						setLineItems: function(modifiedCallBack) {
+							app.models.lineitems
```
Comment (sQcrm) [inline controllers/products.js]: Product does  not have line items so we may not need this function at all

**Review by sQcrm — CHANGES_REQUESTED** [review]
@vivekshekar added some changes needed to be done.

**Comment by vivekshekar** [comment]
@abhikchakraborty Changes are done. 
Please check

File: controllers/products.js (hunk 1 above)
Comment (vivekshekar) [reply controllers/products.js:162]: @abhikchakraborty updated

File: controllers/products.js (hunk 2 above)
Comment (vivekshekar) [reply controllers/products.js]: @abhikchakraborty updated

File: controllers/products.js [hunk 3]
```diff
@@ -126,8 +126,8 @@ module.exports = function(app, config) {
 			},
 			function(err, results) {
 				if (err) return next(err);
-				var productFData = {},
-				productPrefix = results.getProductPrifix;
+				var productFData = {};
+				//productPrefix = results.getProductPrifix;
```
Comment (abhikchakraborty) [inline controllers/products.js]: remove this line.

File: controllers/products.js [hunk 4]
```diff
@@ -153,44 +153,20 @@ module.exports = function(app, config) {
...
-								if (productPrefix && productPrefix.settingData && k === 'product_number') {
-									v= productPrefix.settingData+''+v;
-									productData[k] = v;
-								}
+// 								// prefixing the SO number with the SO prefix from setting
+// 								if (productPrefix && productPrefix.settingData && k === 'product_number') {
+// 									v= productPrefix.settingData+''+v;
+// 									productData[k] = v;
+// 								}
 							});
```
Comment (abhikchakraborty) [inline controllers/products.js:162]: remove the above commented lines.

**Review by abhikchakraborty — COMMENTED** [review]
@vivekshekar tested the changes and looks good. There is a small change request before it could be merged.

**Comment by vivekshekar** [comment]
@abhikchakraborty removed the commented lines.

**Pull request merged**